### PR TITLE
#169257390 Integrate Coveralls test coverage

### PR DESCRIPTION
**Pull request by RaymondMwaura**
#### What does this PR do?
Implements test coverage on the codebase using Coveralls.io

#### Description of Task to be completed?
- [x] install the following dependencies:
   - [x] mocha
   - [x] coveralls
   - [x] nyc
- [x] add further configuration to the 'travis.yml' file to run the 'test' and 'coverage' scripts using 'nyc'
- [x] update the README file with the Coveralls badge
- [x] update '.gitignore' not to track coverage files generated by 'nyc', namely '.nyc_output' 

#### How should this be manually tested?
- After cloning the repo 'cd' into it
- run `npm install` to install all the dependencies
- create a branch from this branch, add some tests to './src/tests'
- run the tests locally by navigating to the project folder using Terminal and running `npm test`
- your tests will either pass or fail and you should see a coverage report printed out on Terminal
- push your branch to Github to trigger a TravisCI build
- after the tests have run you can get on Github, under branches you will be able to see the checks that have been performed on your branch as having passed or failed as well as links to TravisCI and Coveralls respectively. View the coverage report on Coveralls.io 
- once you merge your branch to develop, the README file will refresh after a couple of hours (about 12 hours) to reflect the latest coverage on the newly updated develop branch

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169257390](https://www.pivotaltracker.com/story/show/169257390)

#### Screenshots (if appropriate)
coveralls running
README with Coveralls badge

#### Questions:
N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "nyc mocha src/tests/*.js",
```
Comment (timolinn) [inline package.json]: We'll be using ES6 features, can you add babel? Check #6

**Review by timolinn — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (RaymondMwaura) [reply package.json]: > We'll be using ES6 features, can you add babel? Check #6

Sure. Let me work on that.

Comment (RaymondMwaura) [reply package.json]: I have added babel.

Comment (RaymondMwaura) [reply package.json]: > We'll be using ES6 features, can you add babel? Check #6

I have added babel.

**Review by b0nbon1 — CHANGES_REQUESTED** [review]
I think you should exclude the files that won't be tested like migrations, database, test on `package.json`
by maybe adding `"nyc": {
    "exclude": [
      // files that you to exclude
    ]`

**Comment by RaymondMwaura** [comment]
> I think you should exclude the files that won't be tested like migrations, database, test on `package.json`
> by maybe adding `"nyc": { "exclude": [ // files that you to exclude ]`

The test script already specifies that it targets only JS files in the test folder so it won't target other files.

File: package.json [hunk 2]
```diff
@@ -42,11 +50,19 @@
     "underscore": "^1.9.1"
   },
   "devDependencies": {
+    "@babel/cli": "^7.6.4",
+    "@babel/core": "^7.6.4",
+    "@babel/node": "^7.6.3",
+    "@babel/preset-env": "^7.6.3",
+    "@babel/register": "^7.6.2",
```
Comment (b0nbon1) [inline package.json]: all of this are packages on `dependencies`, why include them on `devDependencies`

Comment (RaymondMwaura) [reply package.json]: Let me fix that.

Comment (timolinn) [reply package.json]: @b0nbon1 I think they are all for development purposes only, so should be in `devDependencies`

Comment (RaymondMwaura) [reply package.json]: I'll move them to devDependencies.

**Review by timolinn — APPROVED** [review]

**Review by b0nbon1 — APPROVED** [review]

**Review by nshaphan — APPROVED** [review]

**Pull request merged**